tests/helper/waf_test.cpp

```diff
@@ -161,6 +163,8 @@ TEST(WafTest, Logging) {
         EXPECT_EQ(sink->count(), 6);
         sink->clear();
     }
+
+    spdlog::set_default_logger(old_logger);
```

Comment:
maybe something like


    struct restore_logger {
        ~restore_logger() { spdlog::set_default_logger(old_logger); }
        std::shared_ptr<spdlog::logger> old_logger = spdlog::default_logger();
    } restorer{};

in the beginning so this is executed even if the test fails/throws

Reply:
Added defer which we'll use in other places as well, thx for the tip.